### PR TITLE
Support Ember v5

### PR DESCRIPTION
**Pull request by jelhan**
This is a breaking change as it is adding `@ember/string` as an additional peer dependency.

It included #550 which should be merged first.

### CODE REVIEW

**Comment by Windvis** [comment]
> This is a breaking change as it is adding @ember/string as an additional peer dependency.

It's just a devDep at the moment, no? Also, @ember/string is a v2 addon so ember-auto-import v2 needs to be a dep I think?

**Comment by jelhan** [comment]
I learned that there is a `2.0.0-beta.2` release, which supports Ember v5. The code is not in this repository. But it is available on NPM.

**Comment by jelhan** [comment]
It should work. Also the added test scenario for Ember v5.12 passes. But I haven't tested in a real app yet.

File: README.md [hunk 1]
```diff
@@ -124,7 +124,7 @@ ember install ember-metrics
 
 - Ember.js v3.20 or above
```
Comment (SergeAstapov) [inline README.md:125]: as a follow up, we may bump support for 3.28+ which should simplify CI and should not affect many (if any) users as 3.20 -> 3.28 upgrade should be pretty simple

**Review by SergeAstapov — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (jelhan) [reply README.md:125]: I have done that in https://github.com/adopted-ember-addons/ember-metrics/pull/549.

**Comment by jelhan** [comment]
It seems that some outdated dependencies weren't compatible with Ember v5 causing an error. After I merged in #550 it works fine. So let's land that upgrade first and than add the support for Ember v5 in the next step.

**Comment by SergeAstapov** [comment]
@jelhan mind to rebase this PR as https://github.com/adopted-ember-addons/ember-metrics/pull/550 landed

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -59,6 +59,7 @@ jobs:
         try-scenario:
           - ember-lts-3.28
           - ember-lts-4.12
```
Comment (SergeAstapov) [inline .github/workflows/ci.yml:61]: @jelhan shouldn't we add the whole matrix of LTS versions in v4 and v5, like 4.4, 4.8, 5.4, 5.8?

Comment (jelhan) [reply .github/workflows/ci.yml:61]: Is there a realistic risk that it works for 4.12 and 5.12 but not for interim releases? CI minutes are free for open source projects on GitHub. But we should nevertheless review the value for each scenario as they have other trade-offs such as environmental impact.

Happy to add if you feel they provide real value.

Comment (SergeAstapov) [reply .github/workflows/ci.yml:61]: @jelhan very good question and considering ember-source had very little changes during v4 and v5, I guess answer is "no". It's more like a convention I've seen around in the ecosystem and been following for the addons I've tried to maintain.

**Pull request merged**